### ISSUE TITLE
zig built from source panics on windows: Progress.zig

### ISSUE THREAD

**Issue by mspiess** [ziglang/zig#8897]
I successfully built zig from source on windows, but whenever I try to build anything with `zig-out/bin/zig.exe` I get the following error:
```
thread 10764 panic: reached unreachable code
C:\Users\redacted\zig\lib\std\Progress.zig:297:13: 0x7ff7d13df80f in std.Progress.refreshWithHeldLock (zig.obj)
            unreachable;
            ^
```

I am using a MSYS2 terminal.

**Comment by LemonBoy** [comment]
Fixed by #8717

**Comment by mspiess** [comment]
I think this bug was introduced by [#8174](https://github.com/ziglang/zig/pull/8174).
Replacing Progress.zig:295
```zig
if (std.builtin.os.tag == .windows) {
```

with
```zig
if (!self.supports_ansi_escape_codes and std.builtin.os.tag == .windows) {
```

seems to fix it.

**Comment by andrewrk** [comment]
@LemonBoy I want to revert the std.Progress changes for an unrelated reason, which is the cursor being left at the beginning rather than the end. Is there any reason to not revert the std.Progress changes? I didn't notice any benefit from #8174. Personally my user experience has only gotten worse since then.

**Comment by andrewrk** [comment]
Fixed by f7ee3b4ca5339791b08d20568270908d5e71e7ea.
